pkg/models/shipment_test.go

```diff
@@ -172,6 +173,28 @@ func (suite *ModelSuite) TestAcceptShipmentForTSP() {
 	suite.Nil(newShipmentOffer.RejectionReason)
 }
 
+// TestCurrentTransportationServiceProviderID tests that a shipment returns the proper current tsp id
+func (suite *ModelSuite) TestCurrentTransportationServiceProviderID() {
+	tsp := testdatagen.MakeTSP(suite.db, testdatagen.Assertions{})
+	shipment := testdatagen.MakeDefaultShipment(suite.db)
+	var emptyUUID uuid.UUID
+
+	suite.Equal(shipment.CurrentTransportationServiceProviderID(), emptyUUID)
+
+	testdatagen.MakeShipmentOffer(suite.db, testdatagen.Assertions{
+		ShipmentOffer: ShipmentOffer{
+			TransportationServiceProviderID: tsp.ID,
+			ShipmentID:                      shipment.ID,
+		},
+	})
+
+	// CurrentTransportationServiceProviderID looks at the shipment offers on a shipment
+	// Since it doesn't re-fetch the shipment, if the offers have changed
+	// We need to re-fetch the shipment to reload the offers
+	reloadShipment, _ := FetchShipmentByTSP(suite.db, tsp.ID, shipment.ID)
```

Comment:
It's worth checking this error (even though it's in a test):

```suggestion
	reloadShipment, err := FetchShipmentByTSP(suite.db, tsp.ID, shipment.ID)
	suite.Nil(err)
```
```